### PR TITLE
💄 Add eslint to the project

### PR DESCRIPTION
**Pull request by mihaerzen**
This PR is related to the issue #7 . It adds `eslint` to the project and fixes all the issues. I do not expect the reviewer to check all the code as it is simply too much, however, I would ask to have a look at the configuration files and let me know if something does not make sense (you can basically skip the `💄 Fix eslint related issues` commit).

Thank you 🖖🏼 

# CLI Usage

`yarn lint`

If you also want to automatically fix the project errors:

`yarn lint --fix`

# IDE Usage

## VSCode

Have a look at step 3 of this article from DigitalOcean -> https://www.digitalocean.com/community/tutorials/linting-and-formatting-with-eslint-in-vs-code

## WebStorm

image

# Remaining issues

There is a leftover with this task. I have added the TODO and will address the remaining `eslint` rules one by one in separate pull requests as the task was getting a bit big.

### CODE REVIEW

**Comment by mihaerzen** [comment]
This PR already passes the "git linting action" I have also added -> https://github.com/sledilnik/zdravniki/actions/runs/1525115762

File: README.md [hunk 1]
```diff
@@ -47,7 +47,7 @@ If you also want to automatically fix the project errors:
 
 #### VSCode
 
-Have a look at step 3 of this article from DigitalOcan -> https://www.digitalocean.com/community/tutorials/linting-and-formatting-with-eslint-in-vs-code
+Have a look at step 3 of [this article from DigitalOcean](https://www.digitalocean.com/community/tutorials/linting-and-formatting-with-eslint-in-vs-code)
```
Comment (mihaerzen) [inline README.md:50]: 👍🏼

**Comment by lukarenko** [comment]
@mihaerzen I tried this PR, did `yarn install`, `yarn lint`, but on `yarn start` I get build issue:
image

**Comment by jalezi** [comment]
It works for me.

@lukarenko yarn install?

**Comment by mihaerzen** [comment]
I'm seeing the same issues actually. Will look into it tomorrow.

**Comment by jalezi** [comment]
My bad. I did't try yarn start.

**Comment by jalezi** [comment]
I partially solved with help from https://githubmemory.com/repo/upleveled/eslint-config-upleveled/issues/49

```javascript

// package.json
"resolutions": {
    "eslint-plugin-react": "7.27.1"
  }
```

Errors not fixed: both rules `import/no-.... was not found`.

**Comment by mihaerzen** [comment]
@lukarenko and @jalezi I have disabled the `eslint` packaged with CRA as it has tons of open issues and it seems it is not really possible to extend it which is a bit of a bummer. Let me know if you have any concerns. Personally, I would merge this ASAP as some issues unearthed by the added linting are pretty nasty...

**Comment by jalezi** [comment]
I would do it asap but maybe @stefanb is the right person to decide.

**Comment by mihaerzen** [comment]
After re-basing and fixing the eslint issues I am getting the following error

image

@jalezi any ideas?

**Comment by jalezi** [comment]
yes. i had same problem. solved with deleting node_modules and yarn install

**Comment by stefanb** [comment]
Unfortunately I don't have the dev environment for this set up yet (likely due to having node 17 from homebrew which seems incompatible with this project, requiring node 16).

I'm fine with whatever works for those that actually develop the project and doesn't interfere too much with the process.

File: .github/workflows/pull-request.yml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: 16
+      - run: npm i -g yarn
+      - run: yarn install --frozen-lockfile
```
Comment (stefanb) [inline .github/workflows/pull-request.yml:14]: If this is too slow we can later cache the `node_modules/` folder between runs.

Comment (mihaerzen) [reply .github/workflows/pull-request.yml:15]: I have added a commit. Let's see if it works.

Comment (stefanb) [reply .github/workflows/pull-request.yml:15]: Wow, i didn'know of this elegant way of caching 😲

Comment (mihaerzen) [reply .github/workflows/pull-request.yml:15]: It seems to be working:
image

Comment (stefanb) [reply .github/workflows/pull-request.yml:15]: Nice, 30 seconds shorter test cycle may add up: 
image
🎉

**Comment by jalezi** [comment]
> Unfortunately I don't have the dev environment for this set up yet (likely due to having node 17 from homebrew which seems incompatible with this project, requiring node 16).
> 
> I'm fine with whatever works for those that actually develop the project and doesn't interfere too much with the process.

I believe I locked node version because of Netlify. They don't support node 17 yet or at least they haven't at the time I started. Also when I started the project I was exploring [nvm](https://github.com/nvm-sh/nvm).

I am pretty sure we can change required node version. In between @stefanb you can temporary omit in `package.json` in `engines` field `node: >=16`
```json
"engines": {
    "node": ">= 16", 
    "npm": "please-use-yarn",
    "yarn": ">=1.22.0"
  }
```

If this is not enough delete `.nmprc` and `.nvmrc`. Am I right @mihaerzen?

**Comment by stefanb** [comment]
Tnx, @jalezi! When @overlordtm arranges PR deployments we can also remove netlify support.

**Comment by jalezi** [comment]
> Tnx, @jalezi! When @overlordtm arranges PR deployments we can also remove netlify support.

I believe we can remove it immediately. I haven't used it since I have transferred ownership. I had to change netlify to point to my fork (which I believe I don't need anymore but this is for another conversation). Since then nothing was deployed while PRs are not pointing to my fork :).

**Comment by stefanb** [comment]
I made it work in my node 17 environment in #60 and then also removed netlify in #61. Please check them.

**Review by stefanb — APPROVED** [review]

**Pull request merged**